app/oauth.go

```diff
@@ -778,8 +778,8 @@ func (a *App) GetAuthorizationCode(w http.ResponseWriter, r *http.Request, servi
 	authURL.RawQuery = queryParams.Encode()
 	authURLString := authURL.String()
 
-	// Following params aren't being "Set" because URLEncode uses '%20' and QueryEscape uses +,
-	// this change "might" break for some SSO Servers + Mattermost.
+	// Following params are being manually appended because URLEncode uses '%20' and QueryEscape uses +,
+	// Using QueryEscape here could be a breaking change for SSO Servers
```

Comment:
I see you've changed the code comment but I'm still a bit fuzzy on this. Could you describe—just here in the pull request for now—in which scenarios this is a breaking change? If you have an example to help illustrate that would be great too.